platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsToolingApiGradleProjectIntegrationTest.groovy

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache.isolated
+
+import org.gradle.tooling.model.GradleProject
+
+class IsolatedProjectsToolingApiGradleProjectIntegrationTest extends AbstractIsolatedProjectsToolingApiIntegrationTest {
+
+    def "can fetch IsolatedGradleProject model"() {
+        settingsFile << """
+            rootProject.name = 'root'
+
+            include("lib1")
+        """
+
+        file("lib1/build.gradle") << """
+            description = "Library 1"
+
+            tasks.register("lazy") {
+                println "realizing lazy task"
+            }
+        """
+
+        when:
+        executer.withArguments(ENABLE_CLI)
+        def projectModels = runBuildAction(new FetchIsolatedGradleProjectForEachProjectInBuild())
+
+        then:
+        outputContains("Running build action to fetch isolated project models")
+
+        projectModels.size() == 2
+        projectModels[0].name == "root"
+        projectModels[0].tasks.size() > 0
+
+        projectModels[1].name == "lib1"
+        projectModels[1].tasks.find { it.name == "lazy" }
+        outputContains("realizing lazy task")
+
+        and: "all properties are available"
+        with(projectModels[1]) {
+            name == "lib1"
+            description == "Library 1"
+            projectIdentifier.projectPath == ":lib1"
+            projectIdentifier.buildIdentifier.rootDir == testDirectory
+            path == ":lib1"
+            buildScript.sourceFile == file("lib1/build.gradle")
+            buildDirectory == file("lib1/build")
+            projectDirectory == file("lib1")
+        }
+
+        when:
+        executer.withArguments(ENABLE_CLI)
+        def projectModels2 = runBuildAction(new FetchIsolatedGradleProjectForEachProjectInBuild())
+
+        then: "build action is cached"
```

Comment:
🤔 It's hard to match this description with assertions below. Maybe `IsolatedProjectsFixture` can be used for that?

Reply:
Good catch! I somehow forgot about the fixture. I now updated all tests to use the fixture 👍